airflow/jobs/base_job.py

```diff
@@ -94,7 +94,8 @@ class BaseJob(Base, LoggingMixin):
 
     def __init__(self, executor=None, heartrate=None, *args, **kwargs):
         self.hostname = get_hostname()
-        self.executor = executor or ExecutorLoader.get_default_executor()
+        if self.__class__.__name__ != "LocalTaskJob":
+            self.executor = executor or ExecutorLoader.get_default_executor()
```

Comment:
The other option is to keep it `None` in `BaseJob` and override it in `SchedulerJob` and `BackfillJob`

Reply:
https://github.com/apache/airflow/pull/16700/commits/a7f6efbabc94512fcf6b8ada80f13b7291a0d2c8 vs https://github.com/apache/airflow/pull/16700/commits/376ffd1deff481717758281608b0764f38753956

Thoughts?

Reply:
another option would be to make `executor` a cached property and `executor_class` a property.  this would simply get them out of init, which i think would be enough if they are not accessed in `LocalTaskJob` anyway.

```python
@cached_property
def executor(self):
    return self._executor or  ExecutorLoader.get_default_executor()  # store init param `executor` as private attr `_executor` 
```

this has two benefits, one is you don't have to reference the subclass name here (as in your first approach) and the other is you don't have to make as many changes re executor in init params)

Reply:
i do also wonder.... why is it that we only get this issue with CKE?  is there perhaps something about the way in which CKE is designed that causes this problem?

Reply:
>i do also wonder.... why is it that we only get this issue with CKE? is there perhaps something about the way in which CKE is designed that causes this problem?

The issue is that it tries to create an instance of `KubernetesExecutor` inside `CeleryExecutor` and `KubernetesExecutor` creates a multiprocessing Manager  & Queue in its `__init__` which creates issues with Celery as explained in https://github.com/celery/celery/issues/4525